rfcs/01780_policy_delegation.md

```diff
@@ -0,0 +1,326 @@
+
+For instance, let us assume the following configuration:
+
+- `https://google.com/oak/keys/count_people_from_pixels.pub`:
+
+  - contents:
+
+    ```text
+    -----BEGIN PUBLIC KEY-----
+    aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+    -----END PUBLIC KEY-----
+    ```
+
+  - used to sign modules with the following hashes:
+    - `0000000000000000000000000000000000000000`
+    - `1111111111111111111111111111111111111111`
+
+- `https://other-company.com/oak/keys/count_people_from_pixels.pub`:
+
+  - contents:
+
+    ```text
+    -----BEGIN PUBLIC KEY-----
+    bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
+    -----END PUBLIC KEY-----
+    ```
+
+  - used to sign modules with the following hashes:
+    - `1111111111111111111111111111111111111111`
+    - `2222222222222222222222222222222222222222`
+
+When an Oak Node with hash `1111111111111111111111111111111111111111` is
+instantiated, it receives the declassification privilege of
+"signature:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa ∧
+signature:bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb ∧
+wasm_hash:1111111111111111111111111111111111111111".
+
+Note that in this case it receives both privileges (or rather, the conjunction
+between these privileges), plus the privilege for the corresponding Wasm hash
+label, which it would get anyways even in the absence of any signed module hash
+referring to it.
+
+In particular, the conjunctive privilege is quite useful in practice as a
+confidentiality label, because it may be used by the data owner to label data
+that is expected to be declassified by a Wasm module that is trusted by multiple
+parties at the same time, where each party independently publishes a separate
+manifest.
+
+## User-based delegation
+
+So far we have talked about companies being the publisher of such manifest, but
+it is possible for individual users to also publish signed module hashes for
+specific use cases. A user may publish a list of signed module hashes on a
+website that they control, or even as part of another website, e.g. on
+github.com. They also need to publish their public key, in a way that allows
+other users to reliably find it and use it as part of an Oak application.
+
+This allows delegation based on existing human-to-human network connections. For
+instance, a person may be an expert in a particular field, and independently
+verify properties of pieces of software and publish them as signed module hashes
+with a particular key, which they would also publish. Also it is possible for a
+person publishing a public key to further delegate to another public key (either
+company or person) by re-signing all the signed module hashes already signed by
+the other key.
+
+For instance, a user may publish a manifest and corresponding signatures at the
+following URL:
+`https://github.com/tiziano88/delegation/count_users_from_pixels.links`:
+
+```json
+{
+  "includes": [],
+  "signatures": [
+    "./count_people_from_pixels_google.sign",
```

Comment:
What do these signature files contain in the case of an expert reviewer? Based on the previous example, I'd expect a public key representing the reviewer. What would the declassification story be in that case? The public key of a reviewer does not represent a principal.